app/scripts/lib/multichain-api/provider-authorize.js

```diff
@@ -68,44 +73,35 @@ export async function providerAuthorizeHandler(req, res, _next, end, hooks) {
   }
 
   try {
+    // use old account popup for now to get the accounts
+    const [subjectPermission] = await hooks.requestPermissions(
+      { origin },
+      {
+        [RestrictedMethods.eth_accounts]: {},
+      },
+    );
+    const permittedAccounts = getAccountsFromPermission(subjectPermission);
     const { flattenedRequiredScopes, flattenedOptionalScopes } = processScopes(
       requiredScopes,
       optionalScopes,
       { findNetworkClientIdByChainId, getInternalAccounts },
     );
 
-    const accounts = [];
     Object.keys(flattenedRequiredScopes).forEach((scope) => {
-      (flattenedRequiredScopes[scope].accounts || []).forEach(
-        (caipAccountId) => {
-          accounts.push(parseAccountId(caipAccountId).address);
-        },
-      );
+      if (scope !== 'wallet') {
```

Comment:
i didnt put `accounts` in the `wallet` scope because im not sure what the caip-10 would be for that